tests/testthat/test-geom-rug.R

```diff
@@ -20,3 +20,10 @@ test_that("coord_flip flips the rugs", {
   expect_equal(length(b[[1]]$children[[1]]$y0), 1)
   expect_equal(length(b[[1]]$children[[1]]$y1), 1)
 })
+
+
+test_that("Rug length needs unit object", {
+  p <- ggplot(df, aes(x,y))
+  expect_is(p + geom_rug(length=grid::unit(0.01, "npc")), "ggplot")
```

Comment:
Could you add some tests about the actual lengths of the grob? As the tests above does, you can check these values:

``` r
library(ggplot2)

n = 10
df <- data.frame(x = 1:n, y = (1:n)^3)

g <- layer_grob(ggplot(df, aes(x, y)) + geom_point() + geom_rug(sides = 'l'), 2)
g[[1]]$children[[1]]$x0
#> [1] 0npc
g[[1]]$children[[1]]$x1
#> [1] 0.03npc
```

<sup>Created on 2019-02-26 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>